testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -4646,6 +4646,49 @@ static void checkRlikeFails(SqlOperatorFixture f) {
         false);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6730">[CALCITE-6730]
+   * Add CONVERT function(enabled in Oracle library)</a>. */
+  @Test void testOracleConvertFunc() {
+    final SqlOperatorFixture f = fixture()
+        .setFor(SqlLibraryOperators.ORACLE_CONVERT, VM_JAVA)
+        .withLibrary(SqlLibrary.ORACLE);
+
+    final Consumer<SqlOperatorFixture> consumer = f0 -> {
+      f0.checkFails("convert('a', utf8, utf10)", "UTF10", false);
+      f0.checkFails("^convert(1, utf8, gbk)^",
+          "Invalid type 'INTEGER NOT NULL' in 'CONVERT' function\\. "
+              + "Only 'CHARACTER' type is supported",
+          false);
+      f0.checkType("convert('a', utf16, gbk)", "CHAR(1) NOT NULL");
+      f0.checkType("convert('a', utf16)", "CHAR(1) NOT NULL");
+      f0.checkType("convert(null, utf16, gbk)", "NULL");
```

Comment:
Maybe we can add test `convert('a', null, gbk)` or `convert('a', gbk, null)`.

Reply:
Oracle's CONVERT doesn't support this, the 2nd or 3rd operand should always be Character set.

Reply:
I also add `convert('a', utf8, ^null^)` and `convert('a', ^null^, utf8)`.